Ensure SdkExt resolvers are properly registered.

pq opened the pull request:
@johnmccutchan @bwilkerson @stereotype441 PTAL.

This could be tightened up (and surely will be once `ContextManager` is doing more of this heavy lifting) but I wanted to make sure I preserved the existing behavior.  In particular, I took pains to ensure that resolvers are added in the same order.  This _may_ be unnecessary but seems safest.  Needless to say, this will get cleaned up majorly down the road.  In the meantime, I hope it gets `SdkExt` support in a world of `.packages` on its feet.  Let me know if you think otherwise!

pq committed:
Ensure SdkExt resolvers are properly registered.

bwilkerson commented:
LGTM

stereotype441 commented on lib/src/driver.dart:
```diff
@@ -388,6 +406,19 @@ class Driver {
+    Map<String, List<fileSystem.Folder>> folderMap =
+        new Map<String, List<fileSystem.Folder>>();
+    packages.asMap().forEach((String path, Uri uri) {
+      folderMap[path] = [PhysicalResourceProvider.INSTANCE.getFolder(uri.path)];
```
This seems incorrect for Windows.  `uri.path` will always be a posix-style path (using forward slashes), whereas PhysicalResourceProvider.INSTANCE.getFolder() expects a native OS path (which on Windows will use backslashes).  I think we need to do `path.toFilePath(Platform.isWindows)`.

stereotype441 replied:
I misspoke in my previous comment: I meant to say `uri.toFilePath(windows: Platform.isWindows)`.

However, on further reflection, I believe it is even better to do `fromUri(uri)` (fromUri() comes from the `path` package).  This is what is done in the analysis server.

stereotype441 committed:
Use path.fromUri() to convert URIs to file paths.

This will perform the appropriate conversions on Windows
(e.g. replacing '/' with '\').

stereotype441 committed:
Fix .packages file discovery on Windows.

The _discoverPackagespec() method was trying to convert its "root"
argument to a directory URI (one ending in a "/") by calling "new
Uri.directory(root.path)".  This was causing confusion on Windows,
since root.path is in posix-style but the Uri.directory constructor
expects a native OS path.

Fortunately, the caller always passes a directory URI anyhow, so no
conversion is necessary.  We can just use the URI as is.

stereotype441 commented:
@bwilkerson PTAL

bwilkerson commented:
LGTM. I assume tests will follow eventually.
